reorder-list/delight010.swift

```diff
@@ -0,0 +1,63 @@
+public class ListNode {
+    public var val: Int
+    public var next: ListNode?
+    public init() { self.val = 0; self.next = nil; }
+    public init(_ val: Int) { self.val = val; self.next = nil; }
+    public init(_ val: Int, _ next: ListNode?) { self.val = val; self.next = next; }
+}
+
+class Solution {
+    // Time O(n)
+    // Space O(1)
+    func reorderList(_ head: ListNode?) {
+        guard let head = head, head.next != nil, head.next?.next != nil else {
+            return
+        }
+        
+        var slow = head
+        var fast = head
+        
+        while fast.next != nil && fast.next?.next != nil {
+            slow = slow.next!
+            fast = fast.next!.next!
+        }
```

Comment:
Optional Chaining 대신 이렇게 쓸 수도 있겠네요
```suggestion
        guard let head = head, let first = head.next, let second = first.next else {
            return
        }

        var slow = head
        var fast = head

        while true {
            guard let first = fast.next, let second = first.next else {
                break
            }
            slow = slow.next!
            fast = second
        }
```